src/main/java/com/solvd/laba/lab1/utils/ConnectionUtil.java

```diff
@@ -14,13 +14,31 @@ public class ConnectionUtil {
     private static final String FILE_PATH = "src/main/resources/properties/database.properties";
     private static final int POOL_SIZE = 7;
     private static BlockingQueue<Connection> connectionPool;
+    private static boolean initialized = false;
 
-    static {
-        initializeConnectionPool();
+    private static synchronized void initializeConnectionPool() {
+        if (!initialized) {
+            connectionPool = new ArrayBlockingQueue<>(POOL_SIZE);
+            createConnection();
+            initialized = true;
+        }
+    }
+
+    public static Connection getConnection() throws InterruptedException {
+        if (!initialized) {
+            initializeConnectionPool();
+        } else if (connectionPool.size() < POOL_SIZE) {
+            //create a connection and add to connection pool
+            createConnection();
```

Comment:
Try to test your connection pool and see if it does limit the connections to 7 and if it does then you're good.
But let's say 10 threads try to get a connection at once. The first one is going to go through the !initialized and the other 9 are going into the else if clause because the size is still less than 7. And because of that there are going to be 9 threads doing createConnection() and you might end up with 10 connections instead of the limit of 7.
If that is true then rework the if else if statement.